contracts/test/integration/IntegrationTestMaplePCVDeposit.t.sol

```diff
@@ -0,0 +1,176 @@
+import {IPool} from "../../pcv/maple/IPool.sol";
+import {DSTest} from "../unit/utils/DSTest.sol";
+import {IDSSPSM} from "../../pcv/maker/IDSSPSM.sol";
+import {Constants} from "../../Constants.sol";
+import {PCVGuardian} from "../../pcv/PCVGuardian.sol";
+import {IMplRewards} from "../../pcv/maple/IMplRewards.sol";
+import {MaplePCVDeposit} from "../../pcv/maple/MaplePCVDeposit.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {PegStabilityModule} from "../../peg/PegStabilityModule.sol";
+import {ERC20CompoundPCVDeposit} from "../../pcv/compound/ERC20CompoundPCVDeposit.sol";
+
+contract IntegrationTestMaplePCVDeposit is DSTest {
+    using SafeCast for *;
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    MaplePCVDeposit private usdcDeposit;
+
+    PCVGuardian private immutable pcvGuardian =
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN);
+
+    Core private core = Core(MainnetAddresses.CORE);
+
+    IERC20 private usdc = IERC20(MainnetAddresses.USDC);
+
+    uint256 public constant targetUsdcBalance = 100_000e6;
+
+    /// @notice once you signal to withdraw after lockup, wait 10 days
+    uint256 public constant cooldownPeriod = 864000;
+
+    /// @notice once you have waited for cool down period to pass
+    /// you have 2 days to withdraw before you have to request to withdraw again
+    uint256 public constant withdrawPeriod = 172800;
+
+    IERC20 public constant maple =
+        IERC20(0x33349B282065b0284d756F0577FB39c158F935e6);
+    address public constant mplRewards =
+        0x7869D7a3B074b5fa484dc04798E254c9C06A5e90;
+    address public constant maplePool =
+        0xFeBd6F15Df3B73DC4307B1d7E65D46413e710C27;
+
+    address public constant mapleOwner =
+        0xd6d4Bcde6c816F17889f1Dd3000aF0261B03a196;
+
+    function setUp() public {
+        usdcDeposit = new MaplePCVDeposit(address(core), maplePool, mplRewards);
+
+        vm.startPrank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        usdc.transfer(address(usdcDeposit), targetUsdcBalance);
+        vm.stopPrank();
+
+        /// governor has pcv controller role
+        vm.prank(MainnetAddresses.GOVERNOR);
+        usdcDeposit.deposit();
+    }
+
+    function testSetup() public {
+        assertEq(address(usdcDeposit.core()), address(core));
+        assertEq(usdcDeposit.balanceReportedIn(), address(usdc));
+        assertEq(address(usdcDeposit.pool()), maplePool);
+        assertEq(address(usdcDeposit.mplRewards()), mplRewards);
+        assertEq(address(usdcDeposit.token()), address(MainnetAddresses.USDC));
+        assertApproxEq(
+            usdcDeposit.balance().toInt256(),
+            targetUsdcBalance.toInt256(),
+            0
+        );
+    }
+
+    function testWithdraw() public {
+        uint256 rewardRate = IMplRewards(mplRewards).rewardRate();
+        vm.prank(mapleOwner);
+        IMplRewards(mplRewards).notifyRewardAmount(rewardRate);
+
+        vm.warp(block.timestamp + IPool(maplePool).lockupPeriod());
```

Comment:
we should experiment and see if we can go 10 days before unlock is true, signal intent to withdraw, wait 10 days, and then withdraw

Reply:
worked :)